src/dissectors/ec_ospf.c

```diff
@@ -140,9 +152,24 @@ FUNC_DECODER(dissector_ospf)
         if (length > buflen)
                 return NULL;
 
-        DISSECT_MSG("OSPF-%s-%d:$netmd5$",
-                ip_addr_ntoa(&PACKET->L3.dst, tmp),
-                ntohs(PACKET->L4.dst));
+        if (auth_data_len == OSPF_AUTH_MD5_SIZE) {
+                DISSECT_MSG("OSPF-%s-%d:$netmd5$",
+                        ip_addr_ntoa(&PACKET->L3.dst, tmp),
+                        ntohs(PACKET->L4.dst));
+        } else if (auth_data_len == OSPF_AUTH_HMAC_SHA1_SIZE) {
+                type = 1;
+        } else if (auth_data_len == OSPF_AUTH_HMAC_SHA256_SIZE) {
+                type = 2;
+        } else if (auth_data_len == OSPF_AUTH_HMAC_SHA384_SIZE) {
+                type = 3;
+        } else if (auth_data_len == OSPF_AUTH_HMAC_SHA512_SIZE) {
+                type = 4;
+        }
+        if (type != 0) {
+                DISSECT_MSG("OSPF-%s-%d:$ospf$%d$",
+                        ip_addr_ntoa(&PACKET->L3.dst, tmp),
+                        ntohs(PACKET->L4.dst), type);
+        }
```

Comment:
Wouldn't this be easier in a *switch/case* statement?
```c
switch (auth_data_len) {
   case OSPF_AUTH_MD5_SIZE:
      DISSECT_MSG(...);
      break;
   case OSPF_AUTH_HMAC_SHA1_SIZE:
   case OSPF_AUTH_HMAC_SHA256_SIZE:
   case OSPF_AUTH_HMAC_SHA512_SIZE:
      DISSECT_MSG(...);
   break;
}
```
A *default*-block with a `return NULL` may even allow to do the check of the first *if*-statement all in one.
This way you also don't require the **type** variable.